enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,181 @@
+---
+title: windows-machine-config-operator
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@openshift-team-cloud"
+  - "@openshift-team-mco"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+replaces:
+  - "/enhancements/windows-containers/ansible-dev-preview.md"
+---
+
+# Windows Machine Config Operator
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows compute node as a day 2 operation with a prescribed configuration to an
+installer provisioned OpenShift 4.6 cluster and enable scheduling of Windows
+workloads. The targeted platforms for 4.6 are vSphere, Azure and AWS. The
+cluster has to be configured with Hybrid OVN networking.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding Windows instances provisioned using the Machine
+  API to OpenShift clusters
+* Perform all the required steps within the VM for it to be added to the cluster
+  as an OpenShift worker node
```

Comment:
both these goals specifically mention "adding" to a cluster, but i am curious if this proposal is meant to cover the entire lifecycle of windows instances? (eg create and destroy)

i see the reference about not covering upgrading/downgrading, but i am more curious about the basic add/remove logic the Machine resources.

Reply:
Removing a Windows worker node from the cluster will be achieved by scaling down the "Windows" MachineSet. LMK if you want me to call that out here.

Reply:
that sounds great. perhaps just change the wording a little in this text, eg:

"Provide workflows for adding and removing Windows instances ..."

"... VM for it to be added or removed from the cluster"

Reply:
I have added your suggestion.